### PR TITLE
Experimental/chrobalt with starboard android

### PR DESCRIPTION
**Pull request by niranjanyardi**
b/368687596

### CODE REVIEW

File: components/embedder_support/android/java/src/org/chromium/components/embedder_support/view/ContentViewRenderView.java [hunk 1]
```diff
@@ -95,6 +95,9 @@ public void surfaceCreated(SurfaceHolder holder) {
                 getSurfaceView().setVisibility(getSurfaceView().getVisibility());
 
                 onReadyToRender();
+
```
Comment (kaidokert) [inline components/embedder_support/android/java/src/org/chromium/components/embedder_support/view/ContentViewRenderView.java:98]: Quick drive by comment - we should `#ifdef` or feature-gate all those changes that change behavior that significantly. 

Unmodified Chromium builds behavior should not change

What is the correct way to `#ifdef` ? I'm not sure, but lets at least pick something, even temporary, to get code landed.

File: third_party/blink/renderer/platform/media/web_media_player_impl.cc [hunk 2]
```diff
@@ -2796,6 +2797,12 @@ std::unique_ptr<media::Renderer> WebMediaPlayerImpl::CreateRenderer(
           &WebMediaPlayerImpl::OnOverlayInfoRequested, weak_this_));
 #endif
 
+  // Always true
```
Comment (kaidokert) [inline third_party/blink/renderer/platform/media/web_media_player_impl.cc:2800]: feature or `#ifdef`-gate this block

File: third_party/blink/renderer/modules/mediasource/media_source.cc [hunk 3]
```diff
@@ -492,13 +492,44 @@ bool MediaSource::isTypeSupported(ExecutionContext* context,
...
+  LOG(INFO) << __func__ << "(" << type << ") -> "
+            << (result ? "true" : "false");
   return result;
 }
 
 // static
 bool MediaSource::IsTypeSupportedInternal(ExecutionContext* context,
                                           const String& type,
                                           bool enforce_codec_specificity) {
+  if (type.Find("99") != kNotFound || type.Find("catavision") != kNotFound ||
```
Comment (kaidokert) [inline third_party/blink/renderer/modules/mediasource/media_source.cc:504]: feature or `#ifdef`-gate this block

File: media/renderers/BUILD.gn [hunk 4]
```diff
@@ -24,6 +24,16 @@ source_set("renderers") {
     "renderer_impl.h",
     "renderer_impl_factory.cc",
     "renderer_impl_factory.h",
+    "sbplayer_bridge.cc",
```
Comment (kaidokert) [inline media/renderers/BUILD.gn:27]: GN code should also be gated on a conditional

File: media/base/android/BUILD.gn [hunk 5]
```diff
@@ -200,6 +200,30 @@ if (is_android) {
       "java/src/org/chromium/media/MediaServerCrashListener.java",
       "java/src/org/chromium/media/ScreenResolutionUtil.java",
       "java/src/org/chromium/media/VideoAcceleratorUtil.java",
+
```
Comment (kaidokert) [inline media/base/android/BUILD.gn:203]: GN code should also be gated on a conditional

File: components/viz/service/display/overlay_processor_android.cc [hunk 6]
```diff
@@ -55,8 +56,8 @@ OverlayProcessorAndroid::OverlayProcessorAndroid(
   // the underlying overlay is opaque anyway; the candidate is referring to
   // a dummy resource that has no relation to what the overlay contains.
   // https://crbug.com/842931 .
-  strategies_.push_back(std::make_unique<OverlayStrategyUnderlay>(
-      this, OverlayStrategyUnderlay::OpaqueMode::AllowTransparentCandidates));
+  strategies_.push_back(std::make_unique<OverlayStrategyUnderlayStarboard>(
```
Comment (kaidokert) [inline components/viz/service/display/overlay_processor_android.cc:59]: `#ifdef` or feature-gate this please

File: components/viz/service/BUILD.gn [hunk 7]
```diff
@@ -338,6 +338,8 @@ viz_component("service") {
       "display/overlay_strategy_single_on_top.h",
       "display/overlay_strategy_underlay.cc",
       "display/overlay_strategy_underlay.h",
+      "display/overlay_strategy_underlay_starboard.cc",
```
Comment (kaidokert) [inline components/viz/service/BUILD.gn:341]: gn code should be gated on a conditional

File: components/embedder_support/android/java/src/org/chromium/components/embedder_support/view/ContentViewRenderView.java (hunk 1 above)
Comment (niranjanyardi) [reply components/embedder_support/android/java/src/org/chromium/components/embedder_support/view/ContentViewRenderView.java:98]: talked about this on chat : https://chromium.googlesource.com/chromium/src/+/HEAD/docs/how_to_add_your_feature_flag.md

this seems like the way to go.

**Pull request closed without merging**